### PR TITLE
[TACHYON-1605] Specify permission required for each filesystem master API

### PR DESCRIPTION
**Pull request by dongche**
JIRA: https://tachyon.atlassian.net/browse/TACHYON-1605

Specify which permissions are needed for each method in FileSystemMaster.
Also add permission check in two updated methods: `getFileInfo`, `getFileInfoList`, whose param is id instead of path before.

@apc999 , @sundapeng

### CODE REVIEW

File: core/server/src/main/java/tachyon/master/file/FileSystemMaster.java [hunk 1]
```diff
@@ -449,6 +454,7 @@ private FileInfo getFileInfoInternal(Inode inode) throws FileDoesNotExistExcepti
       throws AccessControlException, FileDoesNotExistException, InvalidPathException {
     MasterContext.getMasterSource().incGetFileInfoOps(1);
     synchronized (mInodeTree) {
+      checkPermission(FileSystemAction.READ, path, false);
```
Comment (sundapeng) [inline core/server/src/main/java/tachyon/master/file/FileSystemMaster.java]: If the path is a file, requiring `Read` permission is okay, but for a directory, we should check `Execute` first.
I think we should remove the `checkPermission` here (getFileInfoInternal will check it) and add `Execute` permission check at `inode.isDirectory()`

**Comment by sundapeng** [comment]
Thank @dongche for your patch, I left a comment.

**Comment by apc999** [comment]
@dongche @sundapeng , could you resolve the conflicts? The codebase is changed quite a bit during last week. Let me know if you need any help and we could talk offline.

**Comment by apc999** [comment]
@dongche @sundapeng here is a description for more info about the new project name :)
http://www.alluxio.com/2016/02/alluxio-formerly-tachyon-is-entering-a-new-era-with-1-0-release/

**Comment by apc999** [comment]
@dongche do you plan to fix @sundapeng 's comment in this PR?

File: core/server/src/main/java/tachyon/master/file/FileSystemMaster.java (hunk 1 above)
Comment (dongche) [inline core/server/src/main/java/tachyon/master/file/FileSystemMaster.java]: Thanks! Add `Execute` permission check when the path is a dir.

**Comment by dongche** [comment]
@apc999 , @sundapeng , thanks for your review! Patch updated to address comment.

**Comment by sundapeng** [comment]
Thank @dongche . The patch LGTM

**Comment by apc999** [comment]
LGTM

**Pull request merged**